L1Trigger/L1TGlobal/interface/L1TGlobalUtil.h

```diff
@@ -179,35 +175,36 @@ class L1TGlobalUtil {
 
 
     // prescale factors
+    bool m_readPrescalesFromFile;
     const l1t::PrescalesVetosHelper* m_l1GtPrescalesVetoes;
     unsigned long long m_l1GtPfAlgoCacheID;
-    
+
     // prescale or mask algo decisions
     bool m_algorithmTriggersUnprescaled;
-    bool m_algorithmTriggersUnmasked; 
+    bool m_algorithmTriggersUnmasked;
 
     // prescales and masks
     bool m_filledPrescales;
 
     // algorithm maps
     //const AlgorithmMap* m_algorithmMap;
     const std::map<std::string, L1TUtmAlgorithm>* m_algorithmMap;
-    
+
     // Number of physics triggers
     unsigned int m_numberPhysTriggers;
-    
+    const unsigned int m_maxNumberPhysTriggers = 512;
+
     //file  and container for prescale factors
     std::string m_preScaleFileName;
     unsigned int m_PreScaleColumn;
-    
+    unsigned int m_numberOfPreScaleColumns;
+
     std::vector<std::vector<int> > m_initialPrescaleFactorsAlgoTrig;
     const std::vector<std::vector<int> >* m_prescaleFactorsAlgoTrig;
-    std::vector<unsigned int>  m_initialTriggerMaskAlgoTrig;
-    const std::vector<unsigned int>*  m_triggerMaskAlgoTrig;
-    std::vector<int>   m_initialTriggerMaskVetoAlgoTrig;
-    const std::vector<int>*  m_triggerMaskVetoAlgoTrig;
-    
-    // access to the results block from uGT 
+    const std::map<int, std::vector<int> > m_initialTriggerMaskAlgoTrig;
+    const std::map<int, std::vector<int> >*  m_triggerMaskAlgoTrig; // vector stores the BX
```

Comment:
why a pointer? (actually all of these being pointers seems to cause considerable problems in code clarify below)